src/flask/cli.py

```diff
@@ -709,7 +706,7 @@ def convert(self, value, param, ctx):
         except click.BadParameter:
             value = click.STRING(value, param, ctx).lower()
 
-            if value == "adhoc":
+            if _isAdHoc(value):
```

Comment:
camelCase has no place in Python code (see PEP8). Also, you replace a very simple check with a function call where you now need to check the function first to see what it does!